openedx/core/djangoapps/notifications/events.py

```diff
@@ -46,20 +46,32 @@ def notification_event_context(user, course_id, notification):
     }
 
 
-def notification_preferences_viewed_event(request, course_id):
+def notification_preferences_viewed_event(request, course_id=None):
     """
     Emit an event when a user views their notification preferences.
     """
+    event_data = {
+        'user_id': str(request.user.id),
+        'course_id': None,
+        'user_forum_roles': [],
+        'user_course_roles': [],
+        'type': 'account'
+    }
+    if not course_id:
+        tracker.emit(
+            NOTIFICATION_PREFERENCES_VIEWED,
+            event_data
+        )
+        return
     context = contexts.course_context_from_course_id(course_id)
     with tracker.get_tracker().context(NOTIFICATION_PREFERENCES_VIEWED, context):
+        event_data['course_id']: str(course_id)
+        event_data['user_forum_roles'] = get_user_forums_roles(request.user, course_id)
+        event_data['user_course_roles'] = get_user_course_roles(request.user, course_id)
+        event_data['type'] = 'course'
         tracker.emit(
             NOTIFICATION_PREFERENCES_VIEWED,
```

Comment:
nit:
```
    event_data = {
        'user_id': str(request.user.id),
        'course_id': None,
        'user_forum_roles': [],
        'user_course_roles': [],
        'type': 'account'
    }
    if course_id:
        context = contexts.course_context_from_course_id(course_id)
        event_data['course_id']: str(course_id)
        event_data['user_forum_roles'] = get_user_forums_roles(request.user, course_id)
        event_data['user_course_roles'] = get_user_course_roles(request.user, course_id)
        event_data['type'] = 'course'

    with tracker.get_tracker().context(NOTIFICATION_PREFERENCES_VIEWED, context):
        tracker.emit(...
```

Reply:
I don't think it can be done because context needs course_id  and 
`tracker.get_tracker().context(NOTIFICATION_PREFERENCES_VIEWED, context)` cannot be called with None context.